examples/webgl-vector-layer.js

```diff
@@ -0,0 +1,68 @@
+import GeoJSON from '../src/ol/format/GeoJSON.js';
+import Layer from '../src/ol/layer/Layer.js';
+import Map from '../src/ol/Map.js';
+import OSM from '../src/ol/source/OSM.js';
+import TileLayer from '../src/ol/layer/WebGLTile.js';
+import VectorSource from '../src/ol/source/Vector.js';
+import View from '../src/ol/View.js';
+import WebGLVectorLayerRenderer from '../src/ol/renderer/webgl/VectorLayer.js';
+import {
+  DefaultAttributes,
+  packColor,
+} from '../src/ol/renderer/webgl/shaders.js';
+import {asArray} from '../src/ol/color.js';
+
+class WebGLLayer extends Layer {
+  createRenderer() {
+    return new WebGLVectorLayerRenderer(this, {
+      className: this.getClassName(),
+      polygonShader: {
+        attributes: {
+          [DefaultAttributes.COLOR]: function (feature, properties) {
+            const color = asArray(properties.COLOR || '#eee');
+            color[3] = 0.85;
+            return packColor(color);
+          },
+          [DefaultAttributes.OPACITY]: function () {
+            return 0.6;
+          },
+        },
+      },
+      lineStringShader: {
+        attributes: {
+          [DefaultAttributes.COLOR]: function (feature, properties) {
```

Comment:
Wouldn't it make more sense to have a `function (feature, resolution)` here, to allow for resolution based styling? Or at least a `function (feature)`, because the properties could be retrieved from the feature with e.g. `feature.get('COLOR')`? Or are the `properties` different from the feature's properties?

Reply:
Having `resolution` here wouldn't be really useful since this is only evaluated when the webgl buffers are rebuilt (typically on source change); instead, the `u_resolution` uniform is available in the shaders.

As for the `properties` object, it's there to save some time by doing `properties['my-prop']` instead of `feature.get('my-prop')`. Maybe this can be made more explicit?

Reply:
Ok, thanks for the explanation. I added another in-line comment to explain why I don't think that passing the properties as second argument makes much sense. As for the resolution, i think it would make sense to have an example that shows how to use that, even if it is just for now until we have a better API. If users can see how to do property and resolution based styling, we've already achieved a lot.

Reply:
I pushed a change that removes the `properties` argument from the attribute getter.  I agree it will be nice to do resolution-dependent styling, but since that will require customization to the shader (instead of just calculating an attribute value in JS), I think we should wait until we have support for compiling shaders from style expressions.